### PR TITLE
PR 13757 follow-up: add missing with-discountOverlaps Similarity constructor variants (#13845)

### PR DESCRIPTION
**Pull request by cpoerschke**
(cherry picked from commit dab731175c86cc25741f37845136d38a69a9d165)

Resolved Conflicts:
	lucene/CHANGES.txt

### CODE REVIEW

File: lucene/CHANGES.txt [hunk 1]
```diff
@@ -47,6 +52,9 @@ API Changes
   the entire segment should be scored. Subclasses that override the method should instead override its replacement.
   (Luca Cavanna)
 
+* GITHUB#13757: For similarities, provide default computeNorm implementation and remove remaining discountOverlaps setters.
+  (Christine Poerschke, Adrien Grand, Robert Muir)
```
Comment (javanna) [inline lucene/CHANGES.txt:56]: I am afraid is is now too late to modify the changelog for 9.12, as it's already been released?

**Comment by javanna** [comment]
@cpoerschke would you mind getting this in, but backing out the changes to 9.12.0 section of the change log?

File: lucene/CHANGES.txt (hunk 1 above)
Comment (cpoerschke) [reply lucene/CHANGES.txt:56]: Yes, https://lucene.apache.org/core/9_12_0/changes/Changes.html cannot be modified but my understanding is that the addition of the missing entry here would be included in https://lucene.apache.org/core/9_12_1/changes/Changes.html in future then, with the 9.12.1 file including 9.12.0 and earlier content also. Whether or not that's worthwhile, hmm, subjective I guess.

File: lucene/CHANGES.txt [hunk 2]
```diff
@@ -47,6 +52,9 @@ API Changes
...
+
```
Comment (cpoerschke) [inline lucene/CHANGES.txt:57]:
```suggestion
```

File: lucene/CHANGES.txt (hunk 1 above)
Comment (cpoerschke) [reply lucene/CHANGES.txt]: https://github.com/apache/lucene/pull/13891/commits/67b95283137fa6169e4898a6cec5655395cf5213 to remove it.

**Review by ChrisHegarty — APPROVED** [review]

**Pull request merged**